backend/src/server/routes/v1/identity-ldap-auth-router.ts

```diff
@@ -135,19 +135,43 @@ export const registerIdentityLdapAuthRouter = async (server: FastifyZodProvider)
         })
       }
     },
-    preValidation: passport.authenticate("ldapauth", {
-      failWithError: true,
-      session: false
-    }) as any,
-
-    errorHandler: (error) => {
-      if (error.name === "AuthenticationError") {
-        throw new UnauthorizedError({ message: "Invalid credentials" });
-      }
+    preValidation: [
+      async (req, res) => {
+        const { lock } = await server.services.identityLdapAuth.checkLdapLockout({
+          identityId: req.body.identityId,
+          username: req.body.username
+        });
 
-      throw error;
-    },
+        try {
+          const passportRes = await (
```

Comment:
Instead of multiple disconnected services for a single function = there will one single service which has a callback.
That callback will then call the passport authenticate if it success rest of things and fails rest of things.